runtime/src/debug/cpp/KDebug.cpp

```diff
@@ -64,16 +64,16 @@ constexpr int runtimeTypeAlignment[] = {
 extern "C" {
 
 // Buffer that can be used by debugger for inspections.
-RUNTIME_USED char* Konan_DebugBuffer() {
+RUNTIME_USED RUNTIME_WEAK char* Konan_DebugBuffer() {
   return debugBuffer;
 }
 
-RUNTIME_USED int Konan_DebugBufferSize() {
+RUNTIME_USED RUNTIME_WEAK int Konan_DebugBufferSize() {
   return sizeof(debugBuffer);
 }
 
 // Auxilary function which can be called by developer/debugger to inspect an object.
-RUNTIME_USED KInt Konan_DebugObjectToUtf8Array(KRef obj, char* buffer, KInt bufferSize) {
+RUNTIME_USED RUNTIME_WEAK KInt Konan_DebugObjectToUtf8Array(KRef obj, char* buffer, KInt bufferSize) {
```

Comment:
Debugger still uses this API, so now it allows an object of one runtime to leak into other runtime.

Reply:
Now it's up to bindings if use legacy API or rely upon the new API. So this is runtime part of the mechanism.

Reply:
Legacy API turns out to be unsafe.
Why not make it safe by implementing public API through function pointers from  "extended info"?